kornia/utils/image.py

```diff
@@ -76,6 +77,58 @@ def _to_bchw(tensor: torch.Tensor, color_channel_num: Optional[int] = None) -> t
     return tensor
 
 
+def _to_bcdhw(tensor: torch.Tensor, color_channel_num: Optional[int] = None) -> torch.Tensor:
+    """Convert a PyTorch tensor image to BCDHW format.
+
+    Args:
+        tensor (torch.Tensor): image of the form :math:`(D, H, W)`, :math:`(C, D, H, W)`, :math:`(D, H, W, C)` or
+            :math:`(B, C, D, H, W)`.
+        color_channel_num (Optional[int]): Color channel of the input tensor.
+            If None, it will not alter the input channel.
+            If set, it will move the color channel towards channel first input style.
+
+    Returns:
+        torch.Tensor: input tensor of the form :math:`(B, C, D, H, W)`.
+    """
+    if not torch.is_tensor(tensor):
+        raise TypeError(f"Input type is not a torch.Tensor. Got {type(tensor)}")
+
+    if len(tensor.shape) > 5 or len(tensor.shape) < 3:
+        raise ValueError(f"Input size must be a three, four or five dimensional tensor. Got {tensor.shape}")
+
+    if len(tensor.shape) == 3:
+        tensor = tensor.unsqueeze(0)
+
+    if len(tensor.shape) == 4:
+        tensor = tensor.unsqueeze(0)
+
+    if color_channel_num is not None and color_channel_num != 1:
+        channel_list = [0, 1, 2, 3]
+        channel_list.insert(1, channel_list.pop(color_channel_num))
+        tensor = tensor.permute(*channel_list)
+    return tensor
+
+
+def _validate_input_dtype(input: torch.Tensor, accepted_dtypes: List,
+                          min: Optional[float] = None, max: Optional[float] = None) -> None:
```

Comment:
I suggest to add this functionality into `kornia.testing`, rename and split into to `check_dtype` , `check_min_max` (or similar).

Reply:
besides, not sure about the cost of computing min max here how much will affect to the total cost of the function

Reply:
besides, would be also great to include `kornia.testing.check_shape` to raise errors with undesired shapes list